### PR TITLE
fix: no warning when importing unrequested adapter

### PR DESCRIPTION
**Pull request by betodealmeida**
This PR changes `connect` so that when a list of `adapters` is passed, we only try to load adapters from that list.

It also fixes a type error introduced in https://github.com/betodealmeida/shillelagh/pull/187.

Closes https://github.com/betodealmeida/shillelagh/issues/194.

### CODE REVIEW

File: src/shillelagh/backends/apsw/db.py [hunk 1]
```diff
@@ -500,17 +500,20 @@ def connect(
         [(1.0, 2.0), (3.0, 4.0)]
 
     """
+    if adapters is None and safe:
```
Comment (cancan101) [inline src/shillelagh/backends/apsw/db.py:503]: I would say the control flow around safe is a bit hard to follow .

Comment (betodealmeida) [reply src/shillelagh/backends/apsw/db.py:503]: Thanks for the feedback! Yeah, I agree it could be improved. It's also not perfect, as someone might be able to monkeypatch an unsafe adapter to mark it as safe. It probably need some refactoring.

BTW, did you get my email? Sometimes my email gets classified as spam because I self host it.
